packages/react-docs/UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,264 @@
+
+### Split
+- Removed prop `gutter` in favor of `hasGutter` [(#4014)](https://github.com/patternfly/patternfly-react/pull/4014)
+
+### Stack
+- Removed prop `gutter` in favor of `hasGutter` [(#4014)](https://github.com/patternfly/patternfly-react/pull/4014)
+
+### Tabs
+- **Tab:**
+  - The title should be wrapped with `<TabTitleText>` for proper styling.  If you would like to place an Icon in the Tab, it should be wrapped with `<TabTitleIcon>` for proper styling. [(#4146)](https://github.com/patternfly/patternfly-react/pull/4146)
+- **Tabs**:
+  - Changed prop `variant` for consistency. You will need to update all instances of `variant` prop to `component` [(#4146)](https://github.com/patternfly/patternfly-react/pull/4146)
+  - Changed `TabVariant` enum Name to `TabComponent` for consistent naming.  You will now need to update all instances of `TabVariant` to `TabComponent`. [(#4146)](https://github.com/patternfly/patternfly-react/pull/4146)
+
+### Table
+- Updated onSelect type from `(event: React.ChangeEvent<HTMLInputElement>) => void` to `(event: React.FormEvent<HTMLInputElement>) => void` [(#3296)](https://github.com/patternfly/patternfly-react/pull/3296)
+- Removed `cellHeightAuto` transformer. All instances of it should be removed from your application. [(#4170)](https://github.com/patternfly/patternfly-react/pull/4170)
+- `cellWidth('max')` has been replaced with `cellWidth(100)` [(#4170)](https://github.com/patternfly/patternfly-react/pull/4170)
+
+### Text input
+- Removed `isValid` prop, `validated` prop should be used instead. To set a the input to invalid, set `validated` prop to `error` or the  enum value `ValidatedOptions.error`. [(#3975)](https://github.com/patternfly/patternfly-react/pull/3975)
+
+### Text area
+- Removed `isValid` prop, `validated` prop should be used instead. To set a the input to invalid, set `validated` prop to `error` or the  enum value `ValidatedOptions.error`. [(#3975)](https://github.com/patternfly/patternfly-react/pull/3975)
+
+### Title
+- Make `headingLevel` required and remove default of `h2` to improve accessibility [(#3922)](https://github.com/patternfly/patternfly-react/pull/3922)
+- Removes enum `TitleLevel`. Use one of the string values `h1`, `h2`, `h3`, `h4`, `h5`, or `h6` instead.
+- Heading levels should be set contextually and and always be increasing or decreasing by 1 when read top-to-bottom in the DOM.
+- Rename enum `TitleSize` to `TitleSizes` [(#3922)](https://github.com/patternfly/patternfly-react/pull/3922)
+- Removes invalid values `xs` and `sm` for `size` prop. Use size `md` instead.
+
+### Toolbar (formerly DataToolbar)
+- The old Toolbar component that is normally used in the PageHeader has been removed in favor of `PageHeaderTools`. [(#4223)](https://github.com/patternfly/patternfly-react/pull/4223)
```

Comment:
I've been updating our component names to use lowercase, but I've just realized it's a little inconsistent to say `toolbar` but then `PageHeader` and `PageHeaderTools`.

Maybe something to reconsider.

Reply:
We can totally reconsider! I put lowercase because we have them that way in most PatternFly design docs. It made sense to me since they're not proper nouns and remembering to capitalize them every time could be tedious. But if we want to capitalize in cases like this, that works for me! We can always reinvent rules :D